server/publications/collections/product.js

```diff
@@ -53,42 +54,56 @@ export function findProductMedia(publicationInstance, productIds) {
  */
 Meteor.publish("Product", function (productId) {
   check(productId, Match.OptionalOrNull(String));
+
   if (!productId) {
     Logger.debug("ignoring null request on Product subscription");
     return this.ready();
   }
+
+  // verify that parent shop is ready
+  const shop = Reaction.getCurrentShop();
+  if (typeof shop !== "object") {
+    return this.ready();
+  }
+
+
   let _id;
+  let productShopId;
+
+  // Default selector. Any changes come via specific roles
   const selector = {
     isDeleted: { $in: [null, false] },
     isVisible: true
   };
-  let productShopId;
-  const shop = Reaction.getCurrentShop();
-  // verify that parent shop is ready
-  if (typeof shop !== "object") {
-    return this.ready();
+
+  // Permits admins to view both visible and invisible products.
```

Comment:
@prosf @lcampanis 
I added this section back into the code. I'm still unclear as to why it was removed, but tests fail without it. Specifically the tests that ensure that an owner/admin/createProduct user can view hidden products.